supplier_calendar/models/stock_rule.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2020 ForgeFlow S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+
+from odoo import models, api, fields
+from datetime import datetime
+
+class StockRule(models.Model):
+    _inherit = 'stock.rule'
+
+    def _get_purchase_order_date(self, product_id, product_qty, product_uom,
+                                 values, partner, schedule_date):
+        res = super(StockRule, self)._get_purchase_order_date(product_id,
```

Comment:
non blocking point : you can replace by : 

``res = super()._get_purchase_order_date(product_id,``